core/src/test/java/io/questdb/griffin/OrderByAdviceTest.java

```diff
@@ -1186,4 +1186,111 @@ public void testDistinctWithOrderByAnotherColumn() throws Exception {
             TestUtils.assertContains(e.getFlyweightMessage(), "ORDER BY expressions must appear in select list.");
         }
     }
+
+    @Test
+    public void testOrderByPriority_columnNameLiteral() throws Exception {
+        assertQuery("a\n1\n2\n3\n4\n5\n6\n7\n8\n9\n",
+                    "select a from x order by a asc, a desc;",
+                    "create table x as (" +
+                        "select" +
+                                " x a," +
+                                " x % 3 b" +
+                                " from long_sequence(9)" +
+                                ")",
+                    null,
+                    true, true, true);
+    }
+
+    @Test
+    public void testOrderByPriority_columnNameLiteral_differentCases() throws Exception {
+        assertQuery("a\n1\n2\n3\n4\n5\n6\n7\n8\n9\n",
+                "select a from x order by a asc, A desc;",
+                "create table x as (" +
+                        "select" +
+                        " x a," +
+                        " x % 3 b" +
+                        " from long_sequence(9)" +
+                        ")",
+                null,
+                true, true, true);
+    }
+
+    @Test
+    public void testOrderByPriority_columnPositions() throws Exception {
+        assertQuery("a\n1\n2\n3\n4\n5\n6\n7\n8\n9\n",
+                "select a from x order by 1 asc, 1 desc;",
+                "create table x as (" +
+                        "select" +
+                        " x a," +
+                        " x % 3 b" +
+                        " from long_sequence(9)" +
+                        ")",
+                null,
+                true, true, true);
+    }
+
+    @Test
+    public void testOrderByPriority_columnAliases() throws Exception {
+        assertQuery("aliased\n1\n2\n3\n4\n5\n6\n7\n8\n9\n",
```

Comment:
It's also possible to make several aliases for the same column and use them in order by,e.g. 
```sql
select a as a1, a as a2 from x order by a1 asc, a2 desc 
```
but I don't think it matters much here because different literals won't clashm only produce unnecessary comparisons .